Add "eval", "tool", and "config" fields to start message/response (#231)
gradbench/gradbench: README.md

The session proceeds over a series of _rounds_, driven by the eval. In each round, the eval sends a _message_ with a unique `"id"`, and the tool sends a _response_ with that same `"id"`. The message also includes a `"kind"`, which has four possibilities:

> 1. `"kind": "start"` - the eval always sends this message first, waiting for the tool's response to ensure that it is ready to receive further messages. This message and the tool response also contains the name of the eval and tool, and optionally a `"config"` field that contains arbitrary information about how the tool or eval has been configured. This information can be used by programs that do offline processing of log files, but is not otherwise significant to the protocol.

Reviewer: ```suggestion
1. `"kind": "start"` - the eval always sends this message first, waiting for the tool's response to ensure that it is ready to receive further messages. This message may optionally contain the `"eval"` name, and the response may optionally contain the `"tool"` name and/or a `"config"` field that contains arbitrary information about how the tool or eval has been configured. This information can be used by programs that do offline processing of log files, but is not otherwise significant to the protocol.
```

interface StartMessage extends Base {
  kind: "start";

>   eval: string;

Reviewer: ```suggestion
  eval?: string;
```

type Message = StartMessage | DefineMessage | EvaluateMessage | AnalysisMessage;

interface StartResponse extends Base {

>   tool: string;

Reviewer: ```suggestion
  tool?: string;
```
